tests/test_bidsRun.py

```diff
@@ -0,0 +1,163 @@
+import logging
+
+import nibabel as nib
+import numpy as np
+import pytest
+
+from rtCommon.bidsCommon import getNiftiData
+from rtCommon.bidsIncremental import BidsIncremental
+from rtCommon.bidsRun import BidsRun
+from rtCommon.errors import MetadataMismatchError
+
+logger = logging.getLogger(__name__)
+
+
+# Test equality check is correct
+def testEq(oneImageBidsI):
+    run1 = BidsRun()
+    run2 = BidsRun()
+
+    run1.appendIncremental(oneImageBidsI)
+    assert run1 != run2
+
+    run2.appendIncremental(oneImageBidsI)
+    assert run1 == run2
+
+    run2._entities['subject'] = "new_subect"
+    assert run1 != run2
+
+
+# Test numIncrementals output is correct
+def testNumIncrementals(oneImageBidsI):
+    run = BidsRun()
+    assert run.numIncrementals() == 0
+
+    NUM_APPENDS = 20
+    for i in range(NUM_APPENDS):
+        run.appendIncremental(oneImageBidsI)
+        assert run.numIncrementals() == i + 1
+
+
+# Test out of bounds values for get incremental
+def testGetOutOfBounds(oneImageBidsI):
+    run = BidsRun()
+
+    NUM_APPENDS = 10
+    for i in range(NUM_APPENDS):
+        run.appendIncremental(oneImageBidsI)
+
+    with pytest.raises(IndexError):
+        run.getIncremental(NUM_APPENDS)
+        run.getIncremental(NUM_APPENDS + 1)
+        run.getIncremental(-1 * NUM_APPENDS)
+        run.getIncremental(-1 * NUM_APPENDS - 1)
```

Comment:
I would think that after the first assertion is hit it would bypass the rest of the instructions. Do all these actually get called?

Reply:
Great catch -- they do not get called.